src/coreclr/vm/threads.cpp

```diff
@@ -2078,9 +2079,11 @@ void ParseDefaultStackSize(LPCWSTR valueStr)
 
 SIZE_T GetDefaultStackSizeSetting()
 {
-    static DWORD s_defaultStackSizeEnv = CLRConfig::GetConfigValue(CLRConfig::INTERNAL_DefaultStackSize);
+    static DWORD s_defaultStackSizeEnv = CLRConfig::GetConfigValue(CLRConfig::EXTERNAL_Thread_DefaultStackSize);
+    static DWORD s_defaultStackSizeProp = Configuration::GetKnobDWORDValue(W("System.Threading.DefaultStackSize"), 0);
 
-    uint64_t value = s_defaultStackSizeEnv ? s_defaultStackSizeEnv : s_defaultStackSizeProperty;
+    uint64_t value = s_defaultStackSizeEnv ? s_defaultStackSizeEnv
```

Comment:
I assume we will also have the official doc updated to describe how this is respected. Basically we start with `DOTNET_Thread_DefaultStackSize`, then `DEFAULT_STACK_SIZE` property, then `System.Threading.DefaultStackSize`.

I think native AOT just remove the `DEFAULT_STACK_SIZE` option, right?

Reply:
Can we delete the undocumented `DEFAULT_STACK_SIZE` property to keep this simple? (If we believe that we need to document breaks of undocumented surface, we can file a breaking change on it. I do not think it is necessary.)

Reply:
It could delete it. Right now ASP.Net is using it though so it would need to be coordinated.

/cc @BrennanConroy @adityamandaleeka

Reply:
Yep, I never had intention of documenting `DEFAULT_STACK_SIZE` option. It doesn't match our naming conventions and nobody should be using it once we have the public mechanism.

I'll do it this way: commit with support for `DEFAULT_STACK_SIZE`, line up a PR that removes it and once this reaches ASP.NET and ASP.NET switches to the public mechanism, we can merge the removal here.

Doc update at https://github.com/dotnet/docs/pull/47575.